include/ginkgo/core/preconditioner/utils.hpp

```diff
@@ -0,0 +1,37 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_PUBLIC_CORE_PRECONDITIONER_UTILS_HPP_
+#define GKO_PUBLIC_CORE_PRECONDITIONER_UTILS_HPP_
+
+
+#include <type_traits>
+
+
+#include <ginkgo/core/preconditioner/isai.hpp>
+
+
+namespace gko {
+namespace preconditioner {
+namespace detail {
+
+
+// true_type if ExplicitType is an instantiation of TemplateType.
+template <typename ExplicitType, template <typename...> class TemplateType>
+struct is_instantiation_of : std::false_type {};
```

Comment:
Instantiation doesn't fit in this context as it is used to describe instances of a class.
What about: `is_parametered_template_of` or `is_typed_version_of`?